lib/assessment.js

```diff
@@ -4,27 +4,61 @@
 // be called with the current element in the array,
 // its index, and the array itself
 // e.g. callback(currentValue, index, array)
+var reduce = function reduce(array, callback, initialValue) {
+  var previousValue;
+  if (initialValue !== undefined) {
+    previousValue = initialValue;
+  } else {
+    array = array.slice();
+    previousValue = array.shift();
+  }
+
+  array.forEach(function(currentValue, index, array) {
+    previousValue = callback(previousValue, currentValue, index, array);
+  });
+
+  return previousValue;
+};
+
+var numbers = [3, 42, 5, 74, 11];
 
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
+var min = function min(previousValue, currentValue, index, array) {
+  return previousValue < currentValue ? previousValue : currentValue;
 };
 
+console.log(reduce(numbers, min));
+
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
-
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
 };
 
+console.log(reduce(numbers, sum));
+
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+    result = result && callback(currentValue, index, array);
+  });
 
+  return result;
+};
```

Comment:
This is actually `every`, not `any`. It returns `true` only if **all** items in the list (array) yield `true` when passed to the callback. Here is an explanation of the steps required to turn `every` into `any`:

The question asks for a function that returns `true` if the callback returns `true` for **any** element in the array. This suggests that we should be using a logical OR (`||`), not a logical AND (`&&`). A logical AND would be suitable for determining whether a callback returns `true` for **all** elements of the array. 

To know what our initial value of `result` should be, we consult the [truth table for the OR operation](https://en.wikipedia.org/wiki/Logical_disjunction#Truth_table). As we see on row 2 of the truth table, `true || false` evaluates to `true`. Therefore, if we initialize `result` to `true`, `any` will return `true` regardless of what the callback returns for each element of the list -- this is not what we want. There's only one more value to consider: `false`.

We see on lines 4 and 3 of the truth table that `false || false` yields `false`, and `false || true` yields `true`. So, if `result` starts out as `false`, it will remain `false` until it is ORed with `true`. This sounds like what we want. Also, once `result` stores `true`, it will remain `true` no matter what we OR it with. This also sounds like what we want here.

So line 42 should read: `var result = false;` and line 45 should read: `result = result || callback(...);`

3/4 credit.